staging/src/k8s.io/api/networking/v1alpha1/well_known_labels.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+const (
+
+	// LabelIPAddressFamily is used to indicate the IP family of a Kubernetes IPAddress.
+	// This label simplify dual-stack client operations allowing to obtain the list of
+	// IP addresses filtered by family.
+	LabelIPAddressFamily = "ipaddress.kubernetes.io/ip-family"
```

Comment:
This is a big performance improvement for clients, because it allows to filter by selector, and makes it super easier ...
it is a kubernetes.io label only, used by the kubernets controllers ... should we validate it? should we reconcile it and fix it if is wrong?

Reply:
I don't think we do that with the other controllers

Reply:
I can add validation in the repair loop and "repair" if necessary

Reply:
If you need efficient selection, why isn't IPFamily a field with a field selector?  And the value is set based on the name at create time and immutable?

Reply:
to be honest, I just took the easiest path to iterate on that, I wasn't considering it at the beginning, but once I started to implement the allocators, it is super useful to select by family.
I will add a TODO to evaluate the need to move it to a field selector,

Reply:
Can field selectors be used in watch?  I have never done that...

We can revisit this and either make it a field selector or a "magic" label or something.  I agree we need to make some guarantees

Reply:
> Can field selectors be used in watch? I have never done that...

AIUI you can select on `metadata.name` to watch a collection for a specific named object. Quite common if I've understood things right.

Reply:
It seems metav1.LabelSelector can not spec field selectors either

Reply:
If we could put field selectors in a metav1.LabelSelector (I know, naming...) then the namespace magic label would not have been needed.